File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/Python/s826491904_comments_removed.py
import sys

class Main:
    def __init__(self):
        self.sc = Scanner()

    def solve(self):
        N = self.sc.nextInt()
        max_val = min_val = self.sc.nextInt()
        for i in range(1, N):
            in_val = self.sc.nextInt()
            max_val = max(max_val, in_val)
            min_val = min(min_val, in_val)
        print(max_val - min_val)

class Scanner:
    def __init__(self):
        self.tokens = iter(sys.stdin.read().split())

    def nextInt(self):
        return int(next(self.tokens))

if __name__ == "__main__":
    Main().solve()  # End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/cot/Java/C/s826491904_comments_removed.c
#include <stdio.h>

int main() {
    solve();
    return 0;
}

void solve() {
    int N;
    scanf("%d", &N);
    int max, min, in;
    scanf("%d", &in);
    max = min = in;
    for (int i = 1; i < N; i++) {
        scanf("%d", &in);
        if (in > max) max = in;
        if (in < min) min = in;
    }
    printf("%d\n", max - min);
}
// End of Code
